### DISCUSSION TITLE
Invalid address

### DISCUSSION THREAD

**Discussion by harrischrisa** [rejetto/hfs#624] [General]
I have used HFS 3 on my Windows PC for a long while successfully. I now get this error
HFS ~ HTTP File Server
© Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
started 6/12/2024, 4:57:10 PM
version 0.52.9
build 2024-06-11T12:54:37.285Z
cwd C:\ProgramData\HFS3
node v18.5.0
platform win32
pid 24488
config config.yaml
HINT: type "help" for help
http serving on any network : 441
uncaught: Error: Invalid socket address
    at new SocketAddress (node:internal/socketaddress:75:21)
    at parseAddress (C:\snapshot\hfs\dist\src\misc.js:105:12)
    at C:\snapshot\hfs\dist\src\misc.js:101:37
    at C:\snapshot\hfs\dist\src\listen.js:288:42
    at Array.filter (<anonymous>)
    at getIps (C:\snapshot\hfs\dist\src\listen.js:288:29)
    at getUrls (C:\snapshot\hfs\dist\src\listen.js:299:35)
    at printUrls (C:\snapshot\hfs\dist\src\listen.js:312:5)
    at C:\snapshot\hfs\dist\src\listen.js:71:5
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_INVALID_ADDRESS'
}
loading plugin antibrute

The server loads but the Home page has an error
Error: Error: Invalid socket address at new SocketAddress (node:internal/socketaddress:75:21) at parseAddress (C:\snapshot\hfs\dist\src\misc.js:105:12) at C:\snapshot\hfs\dist\src\misc.js:101:37 at C:\snapshot\hfs\dist\src\listen.js:288:42 at Array.filter (<anonymous>) at getIps (C:\snapshot\hfs\dist\src\listen.js:288:29) at getUrls (C:\snapshot\hfs\dist\src\listen.js:299:35) at get_status (C:\snapshot\hfs\dist\src\adminApis.js:105:46) at runNextTicks (node:internal/process/task_queues:60:5) at listOnTimeout (node:internal/timers:533:9)

I can access every other page satisfactorily but Home page still has this error.

I got the problem on 0.52.8. I manually installed 0.52.9 over the top and same problem.

What have I done wrong?

Thanks 
Chris

**Comment by rejetto** [comment]
hey Chris, thanks for reporting!
i don't think you did anything wrong.
from what i can see, something is unexpected on an address the system reports to hfs.
sadly, the address causing the problem is not displayed here (because of the error), so i cannot come to a conclusion (from here).
what i can do is to manage this situation better, producing a better message and maybe even avoiding some possible unwanted effect.
I'm including this in the next 0.53 pre-release, which at this point has reached the state of "release candidate", i guess.

**Reply by harrischrisa** [reply]
I just installed 0.53.0-rc11. Same problem
![HFSerror](https://github.com/rejetto/hfs/assets/16251436/e36752d3-91f8-4de4-ae36-8b7ae56537a7)
Chris

PS I can still download file using the server

**Comment by rejetto** [comment]
thank you very much for your quick feedback.
there are 4 cases and i covered only 3 🫠
i'm going to cover the missing case now.
would you care to check a quick update i can provide you here?

**Reply by harrischrisa** [reply]
I can try another update

**Comment by rejetto** [comment]
ok, i replaced rc11 with rc12, that covers your case.
you can just update normally.

**Reply by rejetto** [reply]
i expect you to not have that error on the web,
but instead you should just get a smaller and clearer error message in the console

**Comment by harrischrisa** [comment]
![image](https://github.com/rejetto/hfs/assets/16251436/91e55073-2778-4563-b039-4d1fe12cf108)
0.53.0-rc12 worked

Thanks

**Comment by rejetto** [comment]
cool, thanks.
can you please tell me the error you see in console?

**Reply by harrischrisa** [reply]
20:23:00 HFS ~ HTTP File Server
20:23:00 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
20:23:00 started 6/12/2024, 8:23:00 PM
20:23:00 version 0.53.0-rc12
20:23:00 build 2024-06-12T10:45:23.884Z
20:23:00 cwd C:\ProgramData\HFS3
20:23:00 node v18.5.0
20:23:00 platform win32
20:23:00 pid 14196
20:23:00 config config.yaml
20:23:00 HINT: type "help" for help
20:23:01 http serving on any network : 441
20:23:01 invalid address [object Object]
20:23:01 serving on http://192.168.50.21:441
20:23:01 serving on http://[object Object]:441
20:23:02 loading plugin antibrute
20:23:02 invalid address [object Object]
20:24:17 invalid address [object Object]
20:24:19 invalid address [object Object]
20:24:21 invalid address [object Object]
20:25:59 invalid address [object Object]

Does this help??

**Reply by rejetto** [reply]
a bit, yes.
I'll see what i can do with it

**Comment by rejetto** [comment]
do you happen to have node.js installed?

**Comment by rejetto** [comment]
[https://github.com/user-attachments/files/15804494/hfs-windows.zip](https://github.com/user-attachments/files/15804494/hfs-windows.zip)
in admin/home, you can can right-click on check-for-updates, and paste this link.
It will install this special version: i made a small change, to print more info instead of "object".

**Comment by harrischrisa** [comment]
Here is the output with 0.53.0 rc12.1.
22:10:38 HFS ~ HTTP File Server
22:10:38 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
22:10:38 started 6/12/2024, 10:10:38 PM
22:10:38 version 0.53.0-rc12.1
22:10:38 build 2024-06-12T12:03:50.368Z
22:10:38 cwd C:\ProgramData\HFS3
22:10:38 node v18.5.0
22:10:38 platform win32
22:10:38 pid 7756
22:10:38 config config.yaml
22:10:38 HINT: type "help" for help
22:10:39 http serving on any network : 441
22:10:39 invalid address  [object Object]
22:10:39 serving on http://192.168.50.21:441
22:10:39 serving on http://[object Object]:441
22:10:40 invalid address  [object Object]
22:10:40 loading plugin antibrute

I haven't installed node myself but it may be installed with PhotoStructure!

**Reply by rejetto** [reply]
wow, this is something truly unexected.
it gives me some insight, but not enough.
i made a simple program that prints a lot of network info.
[print-net.zip](https://github.com/user-attachments/files/15805637/print-net.zip)
would you paste here what it prints?

i made it exe for your convenience, to not install node.js, but the source is very basic
```js
console.log(require('os').networkInterfaces())
pauseForUserInput("Press enter to continue")
```

**Comment by harrischrisa** [comment]
I sent an email with the program output

**Reply by rejetto** [reply]
sorry, got no email.
if you got problems with a@rejetto.com you can try rejetto@gmail.com
